main.py

```diff
@@ -96,16 +124,56 @@ def _archive_text_post_without_discussion_tag(self, submission):
         reply = templates.submission_missing_discussion_tag.format(submission.author.name, submission.id)
 
         self._reply_with_sticky_post(submission, reply, True)
+    
+    def _archive_repost(self, submission):
+        reply = templates.submission_repost
+        
+        self._reply_with_sticky_post(submission, reply, True)
 
-    def _reply_with_sticky_post(self, submission, reply_text, remove=False):
+    def _reply_with_sticky_post(self, submission, reply_text, remove, artist_name=None):
         comment = submission.reply(reply_text)
         comment.mod.distinguish("yes", sticky=True)
 
         if remove is True:
             submission.mod.remove()
 
+        #self.database.save_submission(submission)
+    
+    def _repost_process(self, artist_name, song_name, submission):
+        #always ran at same place, so saves some space
         self.database.save_submission(submission)
-
+        
+        return self._process_artist(artist_name.lower(), song_name.lower(), submission.id)
+    
+    def _process_artist(self, artist_name, song_name, postID):
+        lastPosted = self.repostCheck.get_artist_timestamp(artist_name)
+        currentTime = time.time()
+        
+        if lastPosted is None:
+            self.repostCheck.new_entry(artist_name, song_name, currentTime, postID)
+        else:
+            #set repost time according to the rules
+            allowedArtistTime = min(max(lastPosted[1] * 2592000, 604800), 7776000)
+            if (currentTime - allowedArtistTime) > lastPosted[0]:
+                if lastPosted[1] is 0:
+                    #do song test
+                    songPosted = self.repostCheck.search_song(artist_name, song_name)
+
+                    if songPosted is None:
```

Comment:
I suspect one of these branches is wrong as they both execute the same replace_entry function.

Haven't dug into implementation well enough to be certain but I suspect the first should be insert rather than replace entry.

Reply:
If the song has not been posted before but the artist has, the first branch will run.

If the song also has been posted before but it was long enough ago the second branch runs. The replace function is using insert or replace anyway so it does the correct thing

Reply:
I guess what I'm saying is we should probably collapse the if/elif branches.

`if songPosted is None or (currentTime - 604800) > songPosted[0]`

so that we don't have two branches that do the same function call.

Reply:
I wasn't sure if that would error or not, but if it does not error im all for it

Reply:
I tested it, it does error, so it has to stay this way

Reply:
Cancel that, I tested it wrongly, it will not stay this way